### PR TITLE
Fix: Allow indented comments when parsing.

### PR DESCRIPTION
**Pull request by dbanty**
First time using `nom` so let me know if there's a cleaner way to do this.

### CODE REVIEW

**Review by genos — APPROVED** [review]
Nicely done!

**Review by ameyer-rigetti — APPROVED** [review]

File: src/parser/common.rs [hunk 1]
```diff
@@ -217,10 +216,54 @@ pub fn skip_newlines_and_comments<'a>(input: ParserInput<'a>) -> ParserResult<'a
         value((), token!(Comment(v))),
         token!(NewLine),
         token!(Semicolon),
+        preceded(many0(token!(Indentation)), value((), token!(Comment(v)))),
```
Comment (kalzoo) [inline src/parser/common.rs:219]: Looks like this can just replace line 216

**Review by kalzoo — CHANGES_REQUESTED** [review]
Good fix!

File: src/parser/common.rs [hunk 2]
```diff
@@ -217,10 +216,54 @@ pub fn skip_newlines_and_comments<'a>(input: ParserInput<'a>) -> ParserResult<'a
...
     )))(input)?;
     Ok((input, ()))
 }
 
+#[cfg(test)]
+mod describe_skip_newlines_and_comments {
```
[the remark this answers is not in the document]
Comment (dbanty) [reply src/parser/common.rs:225]: Actually it's [Jasmine](https://jasmine.github.io/) influence, but it wouldn't surprise me if Jasmine was inspired by rspec.

**Review by kalzoo — APPROVED** [review]

**Pull request merged**